packages/@aws-cdk/aws-apigateway/lib/restapi.ts

```diff
@@ -204,10 +204,12 @@ export class RestApi extends Resource implements IRestApi {
   private readonly methods = new Array<Method>();
 
   constructor(scope: Construct, id: string, props: RestApiProps = { }) {
-    super(scope, id);
+    super(scope, id, {
+      physicalName: props.restApiName,
+    });
 
     const resource = new CfnRestApi(this, 'Resource', {
-      name: props.restApiName || id,
+      name: this.physicalName.value || id,
```

Comment:
That looks a bit odd. The `|| id` means that the default behavior here is to allocate a physical name if nothing is assigned by the user, which is different than the normal behavior. I believe this is required by the RestAPI resource, and if I remember correctly, this is basically just a display name so doesn't have to be unique in any way or form.

I'd expect `this.physicalName.value` to return this value.

I wonder if the right approach should be to insert the default when calling to `super`:

```ts
super(scope, id, {
  physicalName: props.restApiName || PhysicalName.of(id)
}
```

Reply:
Yes. The problem here is that `this.id` is not available in the `super` call. We can change the value passed to `PhysicalName.of()` to be lazy, but that might actually change the semantic in a few cases (some tests, for example, assume that the `id` is a resolved value - not saying this particular case does it; I'll check).

Reply:
Oh, sorry, I was thinking about a different case, where id = uniqueId. Will change.

Reply:
Done.